### DISCUSSION TITLE
"server closed the connection unexpectedly"

### DISCUSSION THREAD

**Discussion by TobiasSchemenz** [apache/airflow#36689] [Q&A] [answered]
Hi there,

we are using Airflow Version: v2.5.1 in a docker image to run R-scripts using the BashOperator. Generally, this runs perfectly fine, but we have a specific problem we keep running into.
Namely, that (within one task) when we are connecting to a database (in this case, postgres), do something else for at least 5.5 minutes, and then try to connect again to the database, we receive the error in the title, or
```
Error in postgresqlExecStatement(conn, statement, ...) :
RPosgreSQL error: could not Retrieve the result : server closed the connection unexpectedly
   This probably means the server terminated abnormally
   before or while processing the request.
```
This does not happen when we run the exact same script manually within the container — it only happens when Airflow is running it in its BashOperator; and it only happens if the "idle" time between queries is at least 5.5 minutes long.

Did anyone else come across something similar to this issue and has a solution? Or has any idea why this is happening?

**Comment by Taragolis** [comment]
Difficult to say it could be different reason, I pretty sure this one is outside of Airflow itself, just wild guesses:
- Connection issues, for example TCP Keepalive or something simmilar
- Connection closed by the server, e.g.  by `idle_in_transaction_session_timeout` or `statement_timeout`, for more detail better to have a look on [Postgres documentation ](https://www.postgresql.org/docs/current/runtime-config-client.html)

Better to check Postgres logs and configs

**Reply by Taragolis** [reply, marked as answer]
> if the "idle" time between queries is at least 5.5 minutes long.

It is maybe not your case but If you use AWS Cloud and use NAT Gataways, then you might have a look at [Internet connection drops after 350 seconds](https://docs.aws.amazon.com/vpc/latest/userguide/nat-gateway-troubleshooting.html#nat-gateway-troubleshooting-timeout)

**Reply by TobiasSchemenz** [reply]
> > if the "idle" time between queries is at least 5.5 minutes long.
> 
> It is maybe not your case but If you use AWS Cloud and use NAT Gataways, then you might have a look at [Internet connection drops after 350 seconds](https://docs.aws.amazon.com/vpc/latest/userguide/nat-gateway-troubleshooting.html#nat-gateway-troubleshooting-timeout)

You know, that's probably it. Thanks for the answer (and sorry for the late reply)
